rotkehlchen/chain/ethereum/modules/pendle/decoder.py

```diff
@@ -22,6 +45,76 @@ def __init__(
             evm_inquirer=evm_inquirer,
             base_tools=base_tools,
             msg_aggregator=msg_aggregator,
-            ve_pendle_contract=VE_PENDLE_CONTRACT_ADDRESS,
-            pendle_token=PENDLE_TOKEN,
         )
+        CustomizableDateMixin.__init__(self, base_tools.database)
+
+    def _decode_ve_pendle_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] == NEW_LOCK_POSITION_TOPIC:
+            refund_event = None
+            for event in context.decoded_events:
+                # we don't compare the amount because it is the total amount locked that is emitted
+                # and not the amount that was currently deposited to be locked.
+                if (
+                        event.address == VE_PENDLE_CONTRACT_ADDRESS and
+                        event.asset == PENDLE_TOKEN and
+                        event.event_type == HistoryEventType.SPEND and
+                        event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.notes = f'Lock {event.amount} PENDLE in voting escrow until {self.timestamp_to_date(lock_time := deserialize_timestamp(int.from_bytes(context.tx_log.data[32:64])))}'  # noqa: E501
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.extra_data = {'lock_time': lock_time}
+                    event.counterparty = CPT_PENDLE
+                    break
+
+                elif (
+                        event.address == VE_PENDLE_CONTRACT_ADDRESS and
+                        event.asset == A_ETH and
+                        event.event_type == HistoryEventType.RECEIVE and
+                        event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    refund_event = event
+
+                elif (
+                    event.address == VE_PENDLE_CONTRACT_ADDRESS and
+                    event.asset == A_ETH and
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.counterparty = CPT_PENDLE
+                    event.event_subtype = HistoryEventSubType.FEE
+                    event.amount = (amount := event.amount if refund_event is None else event.amount - refund_event.amount)  # noqa: E501
+                    event.notes = f'Pay {amount} ETH as vePendle state broadcast fee'
```

Comment:
those appear always before the lock event?

Reply:
yes.

Reply:
can you have it in a comment? Just for the reader to know